presentation/testing/src/test/IModelUtilities.test.ts

```diff
@@ -2,173 +2,184 @@
+      await buildTestIModel(fileName, () => { });
 
-    await buildTestIModel("name", cb);
+      expect(fake.firstCall.firstArg).to.include(`${fileName}.bim`);
+      expect(fake.firstCall.lastArg).to.deep.equal({ rootSubject: { name: fileName } });
+    });
 
-    expect(cb.calledOnce);
-  });
+    it("builder calls provided callback function", async () => {
+      setupSnapshot();
+      const cb = sinon.spy();
 
-  it("builder saves database changes and closes it when callback succeeds", async () => {
-    setupSnapshot();
+      await buildTestIModel("name", cb);
 
-    await buildTestIModel("name", () => {});
+      expect(cb.calledOnce);
+    });
 
-    expect(snapshotDbMock.verify((x) => x.saveChanges("Created test IModel"), moq.Times.once()));
-    expect(snapshotDbMock.verify((x) => x.close(), moq.Times.once()));
-  });
+    it("builder saves database changes and closes it when callback succeeds", async () => {
+      const snapshotDbMock = setupSnapshot();
 
-  it("builder saves database changes and closes it when callback throws", async () => {
-    setupSnapshot();
-    const cb = () => {
-      throw new Error("TestError");
-    };
+      await buildTestIModel("name", () => { });
 
-    try {
-      await buildTestIModel("name", cb);
-    } catch{}
+      expect(snapshotDbMock.verify((x) => x.saveChanges("Created test IModel"), moq.Times.once()));
+      expect(snapshotDbMock.verify((x) => x.close(), moq.Times.once()));
+    });
 
-    expect(snapshotDbMock.verify((x) => x.saveChanges("Created test IModel"), moq.Times.once()));
-    expect(snapshotDbMock.verify((x) => x.close(), moq.Times.once()));
-  });
+    it("builder saves database changes and closes it when callback throws", async () => {
+      const snapshotDbMock = setupSnapshot();
+      const cb = () => {
+        throw new Error("TestError");
+      };
+
+      try {
+        await buildTestIModel("name", cb);
+      } catch { }
+
+      expect(snapshotDbMock.verify((x) => x.saveChanges("Created test IModel"), moq.Times.once()));
+      expect(snapshotDbMock.verify((x) => x.close(), moq.Times.once()));
+    });
 
-  it("returns result of SnapshotConnection.openFile", async () => {
-    snapshotDbMock = moq.Mock.ofType<SnapshotDb>();
-    snapshotConnectionMock = moq.Mock.ofType<SnapshotConnection>();
-    sinon.stub(SnapshotDb, "createEmpty").returns(snapshotDbMock.object);
-    sinon.stub(SnapshotConnection, "openFile").resolves(snapshotConnectionMock.object);
-    configureForPromiseResult(snapshotConnectionMock);
+    it("returns result of SnapshotConnection.openFile", async () => {
+      const snapshotDbMock = moq.Mock.ofType<SnapshotDb>();
+      const snapshotConnectionMock = moq.Mock.ofType<SnapshotConnection>();
+      sinon.stub(SnapshotDb, "createEmpty").returns(snapshotDbMock.object);
+      sinon.stub(SnapshotConnection, "openFile").resolves(snapshotConnectionMock.object);
+      configureForPromiseResult(snapshotConnectionMock);
```

Comment:
We already have the `setupSnapshot` functon which does similar stuff. But there are a few tests that need slight variations and because of that, they need to duplicate everything the function does.

Maybe `setupSnapshot` could do everything we need in all the tests and return a combination of things:
```ts
interface SetupSnapshotResult {
  dbMock: moq.IMock<SnapshotDb>;
  connectionMock: moq.IMock<SnapshotConnection>;
  createSnapshotDb: // don't know type of sinon.fake
  openSnapshotConnection: // don't know type of sinon.fake
};
```

Then in tests we could take whatever we need, e.g. here:
```ts
const { connectionMock } = setupSnapshot();
const result = await buildTestIModel("name", () => { });
expect(result).to.equal(connectionMock.object);
```

Reply:
Done